### PR TITLE
feat: adding aria-label alias for accessibilityLabel

### PR DESCRIPTION
**Pull request by Viraj-10**
## Summary

- Adding [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) alias for [accessibilityLabel](https://reactnative.dev/docs/accessibility#accessibilitylabel)  

https://github.com/facebook/react-native/issues/34424

## Changelog

[General] [Added] - Add `aria-label` prop to `Button`, `View`, `Pressable` component

## Test Plan
```
<Button
     onPress={() => onButtonPress('cancelled')}
     testID="cancel_button"
     color={theme.SystemRedColor}
     title="Cancel Application"
     aria-lable="Press to cancel your application!"
/>
```

### CODE REVIEW

**Comment by Viraj-10** [comment]
Hi @necolas, I Just Wanted to know if this is the right way. I will add other components if it's good to go. Open to suggestions.

File: Libraries/Components/Button.js [hunk 1]
```diff
@@ -152,6 +152,12 @@ type ButtonProps = $ReadOnly<{|
   importantForAccessibility?: ?('auto' | 'yes' | 'no' | 'no-hide-descendants'),
   accessibilityHint?: ?string,
   accessibilityLanguage?: ?Stringish,
+
+  /**
+   * Web to Native Accessibilty props
+   * https://github.com/facebook/react-native/issues/34424
+   */
+  'aria-lable'?: ?string,
```
Comment (necolas) [inline Libraries/Components/Button.js:160]: The prop name is `aria-label`

File: Libraries/Components/Button.js [hunk 2]
```diff
@@ -332,7 +339,7 @@ class Button extends React.Component<ButtonProps> {
         accessible={accessible}
         accessibilityActions={accessibilityActions}
         onAccessibilityAction={onAccessibilityAction}
-        accessibilityLabel={accessibilityLabel}
+        accessibilityLabel={accessibilityLabel || ariaLable}
```
Comment (necolas) [inline Libraries/Components/Button.js:342]: 'aria-label' should come first

**Review by necolas — CHANGES_REQUESTED** [review]
Also needs tests or RN tester example alongside where ever any `accessibilityLabel` tests/examples live

**Comment by cipolleschi** [comment]
Hi @Viraj-10, could you rebase your branch, please?

**Comment by Viraj-10** [comment]
> Hi @Viraj-10, could you rebase your branch, please?

@cipolleschi, okay

**Comment by cipolleschi** [comment]
@Viraj-10 I have to ask if you can rebase this PR again... Also thanks for applying the suggestions!

File: Libraries/Components/View/ViewPropTypes.js [hunk 3]
```diff
@@ -452,6 +452,12 @@ export type ViewProps = $ReadOnly<{|
    */
   accessibilityHint?: ?Stringish,
 
+  /**
+   * Alias for accessibilityLabel  https://reactnative.dev/docs/view#accessibilitylabel
+   * https://github.com/facebook/react-native/issues/34424
+   */
+  'aria-label'?: ?Stringish,
+
```
Comment (necolas) [inline Libraries/Components/View/ViewPropTypes.js:460]: Type is Stringish here but string in the other components. Should we be using Stringish everywhere?

Comment (Viraj-10) [reply Libraries/Components/View/ViewPropTypes.js:460]: Actually, I am giving types similar to `accessibilityLabel`.

**Comment by Viraj-10** [comment]
> @Viraj-10 I have to ask if you can rebase this PR again... Also thanks for applying the suggestions!

Sure, No worries.

**Comment by cipolleschi** [comment]
There are some errors in CI, could you have a look at them, please?

**Comment by cipolleschi** [comment]
Hi @Viraj-10, there are still errors. You can see them here: https://app.circleci.com/pipelines/github/facebook/react-native/15336/workflows/555b206f-388d-4dbd-868e-4727297563be/jobs/283063

An example of those is:
```
/home/circleci/react-native/Libraries/Components/Pressable/Pressable.js
  226:5  error  Duplicate key 'accessibilityLabel'  no-dupe-keys
  227:5  error  Duplicate key 'accessibilityState'  no-dupe-keys
```

could you have a look at them?

**Comment by Viraj-10** [comment]
Hi @cipolleschi @necolas, Sorry for the delay. I guess I have added an alias in all the components possible. Please review Once and let me know.

File: Libraries/Components/Button.js [hunk 4]
```diff
@@ -153,6 +157,12 @@ type ButtonProps = $ReadOnly<{|
   importantForAccessibility?: ?('auto' | 'yes' | 'no' | 'no-hide-descendants'),
   accessibilityHint?: ?string,
   accessibilityLanguage?: ?Stringish,
+
+  /**
+   * Alias for accessibilityLabel  https://reactnative.dev/docs/view#accessibilitylabel
+   * https://github.com/facebook/react-native/issues/34424
+   */
+  'aria-label'?: ?string,
```
Comment (necolas) [inline Libraries/Components/Button.js:165]: Duplicating type again

File: Libraries/Components/View/View.js [hunk 5]
```diff
@@ -30,6 +30,8 @@ const View: React.AbstractComponent<
 > = React.forwardRef(
   (
     {
+      accessibilityLabel,
+      'aria-label': ariaLable,
```
Comment (necolas) [inline Libraries/Components/View/View.js:34]: ariaLable -> ariaLabel

File: Libraries/Components/View/View.js [hunk 6]
```diff
@@ -114,6 +116,7 @@ const View: React.AbstractComponent<
     return (
       <TextAncestor.Provider value={false}>
         <ViewNativeComponent
+          accessibilityLabel={ariaLable ?? accessibilityLabel}
```
Comment (necolas) [inline Libraries/Components/View/View.js:119]: ariaLable -> ariaLabel

File: Libraries/Components/Button.js (hunk 4 above)
Comment (Viraj-10) [reply Libraries/Components/Button.js:165]: Done.

File: Libraries/Components/View/View.js (hunk 5 above)
Comment (Viraj-10) [reply Libraries/Components/View/View.js]: Corrected.

File: Libraries/Components/View/View.js (hunk 6 above)
Comment (Viraj-10) [reply Libraries/Components/View/View.js]: Corrected.

**Comment by Viraj-10** [comment]
Hi @cipolleschi, All test cases are passed and I have resolved all the merge conflicts.

**Pull request closed without merging**